virttest/env_process.py

```diff
@@ -603,6 +605,10 @@ def preprocess(test, params, env):
             except path.CmdNotFoundError, msg:
                 raise exceptions.TestSkipError(msg.message)
 
+    global _setuper_mgr
+    _setuper_mgr = test_setup.SetuperMgr(test, params, env)
+    _setuper_mgr.do_setup()
```

Comment:
How about to create `SetupMgr` without arguments and add method to set those arguments later? That would avoid the ugly use of `global` (on top of `env_process` it'd become `_setuper_mgr = test_setup.SetuperMgr()` and here only `_setuper_mgr.initialize(test, params, env)`.

Reply:
The `SetupManager` is designed to be a member of `VirtTest`, and yes, why I didn't make it as so is because we don't have the `env` in the init phase of `test`.

I think your suggestion shall make sense, let me have a try.